app/controllers/concerns/foreman/controller/registration_commands.rb

```diff
@@ -16,7 +16,27 @@ def registration_args
   end
 
   def insecure
-    registration_params['insecure'] ? '--insecure' : ''
+    if registration_params['insecure']
+      if download_utility == 'curl'
+        ' --insecure'
```

Comment:
Nit: Space at the beginning should be handled in the `command` method

Reply:
Applied